### PR TITLE
[ntuple] C-style array RField support

### LINKED ISSUE #11732: [ntuple] C-style array members of a user-defined class are stored/retrieved as a scalar
### Describe the bug
Reported by @Nowakus via e-mail: user-defined classes having members whose type is a C-style array of type `T` are stored/retrieved as a scalar.  Thus, only the first element of the array is read/written.

Specifically, for the following member definition:
```c++
unsigned char m_8pack[8];
```
the corresponding attached field is:
```
      Field 1.3.3    : m_8pack (std::uint8_t)
```

### Expected behavior
In principle we should be throwing, as C-style arrays are currently not supported in RNTuple - i.e. we are missing [here](https://github.com/root-project/root/blob/master/tree/ntuple/v7/src/RField.cxx#L830) a
```c++
if (dataMember->Property() & kIsArray)
   throw /*...*/;
```

However, not supporting C-style arrays might be a problem - please, comment here on it @Nowakus.

Also, in principle, it should be fairly straightforward to add support for single dimension C-style arrays, so we probably should go this way in the fix for this issue.

### PR DESCRIPTION
**Pull request by jalopezg-r00t**
This pull request introduces RField support for C style arrays.  In particular, it provides a `RField<T[N]>` template specialization and allows for parsing of types of the form `T[n][m]...` in `RFieldBase::Create()`

Actual RField implementation is provided by RArrayField, which currently does not support multi-dimensional arrays, although it should be trivial to extend it (if need be in a future PR) to support row-major (C order) multi-dimension arrays.

Additionally, prior to this PR, class members whose type was an array of the form `T[n]` were stored as a scalar.  This is considered a bug (see #11732 for all the details).
The implementation provided here also fixes issue #11732.
 
## Changes or fixes:
- Add `RField<ItemT[N]>` template specialization that matches single-dimension arrays.
- Update `RFieldBase::Create()` to parse types of the form `T[n][m]...`.  As of this PR, it returns `R__FAIL()` for multi-dimensional arrays, though it should be trivial to add support for row-major multi-dimension arrays. 
- RClassField: for class members whose type is a C style array, complete the type name with the size for each dimension, e.g. `int[4][2]`
- Add array unit tests

## Checklist:
- [X] tested changes locally
- [X] updated the docs (if necessary)

This PR fixes #11732.

### CODE REVIEW

**Review by jblomer — CHANGES_REQUESTED** [review]
Great!  Please just update the specification.md, too.

File: tree/ntuple/v7/src/RField.cxx [hunk 1]
```diff
@@ -173,6 +200,14 @@ ROOT::Experimental::Detail::RFieldBase::Create(const std::string &fieldName, con
 
    std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> result;
 
+   if (auto [arrayBaseType, arraySize] = ParseArrayType(typeName); !arraySize.empty()) {
```
[the remark this answers is not in the document]
Comment (jalopezg-r00t) [reply tree/ntuple/v7/src/RField.cxx]: This was actually a "typo"; the intended thing to use here was `normalizedType`, indeed!  Thanks for pointing it out!  :+1:

> Also, I guess type name normalization could also take care of removing whitespaces (not for this PR), which is then useful to correctly parse the array type.

It actually already does, via `TClassEdit::CleanType()` (and also resolves any pending `typedef`s).  I also updated the tests to include some whitespace.

**Review by jblomer — APPROVED** [review]
